File: frontend/src/types/competency.ts
// TSS PPM v3.0 - Competency Types

export interface Competency {
  id: string
  level: string
  category: 'Dedicated' | 'Entrepreneurial' | 'Innovative'
  subcategory: string
  title_en: string
  title_nl?: string
  title_es?: string
  indicators_en?: string[]
  display_order: number
}

export interface CompetencyScore {
  competency_id: string
  score: number | null
  notes?: string
}

export type TovLevel = 'A' | 'B' | 'C' | 'D'

File: backend/src/schemas/competency.py
# TSS PPM v3.0 - Competency Schemas
"""Pydantic schemas for competency data."""

from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CompetencyResponse(BaseModel):
    """Schema for competency response."""

    id: UUID
    level: str
    category: str
    subcategory: str
    title_en: str
    title_nl: Optional[str] = None
    title_es: Optional[str] = None
    indicators_en: Optional[List[str]] = None
    display_order: int

    model_config = ConfigDict(from_attributes=True)
